main/src/use_case/add_friend/AddFriendInteractor.java

```diff
@@ -3,37 +3,35 @@
 import entity.CommonUser;
 import entity.User;
 
+import java.time.LocalDateTime;
+
 public class AddFriendInteractor implements AddFriendInputBoundary{
 
     final AddFriendUserDataAccessInterface userDataAccessInterface;
     final AddFriendOutputBoundary userPresenter;
     final User user;
-    final User userToAdd;
 
     public AddFriendInteractor(AddFriendUserDataAccessInterface AddFriendDataAccessInterface,
                                AddFriendOutputBoundary AddFriendOutputBoundary,
-                               User user, User userToAdd) {
+                               User user) {
         this.userDataAccessInterface = AddFriendDataAccessInterface;
         this.userPresenter = AddFriendOutputBoundary;
         this.user = user;
-        this.userToAdd = userToAdd;
     }
 
 
     @Override
     public void execute(AddFriendInputData addFriendInputData) {
-
+        User userToAdd = userDataAccessInterface.get(addFriendInputData.getFriendUsername());
```

Comment:
We don't know that the friend exists yet as you check for whether they exist in the following if statement. This can produce an error.